### PR TITLE
Common prelimit counter

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Performance Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Use single row counter for multiple streams in pre-limit transform. This helps to avoid uniting pipeline streams in queries with `limit` but without `order by` (like `select f(x) from (select x from t limit 1000000000)`) and use multiple threads for further processing.

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]
Ok. Only a comment before LimitState remain.

**Comment by alexey-milovidov** [comment]
This is related: https://clickhouse-test-reports.s3.yandex.net/9602/aa72aa540cbc0e2d1c149e682d556463a700c2d9/functional_stateless_tests_(debug)/test_run.txt.out.log

**Comment by KochetovNicolai** [comment]
Some queries hang in debug. Need to investigate.

**Comment by KochetovNicolai** [comment]
The query
```
SELECT *
FROM system.numbers_mt
WHERE concat(materialize('1'), '...', toString(number)) = '1...10000000'
LIMIT 1

┌───number─┐
│ 10000000 │
└──────────┘
↗ Progress: 3.02 billion rows, 24.20 GB (323.60 million rows/s., 2.59 GB/s.) Cancelling query.
```
always hang. It is because there is single row which apply to `where` condition.
I think it's possible to unite all limit transforms to big one, with N inputs and N outputs.

**Comment by alexey-milovidov** [comment]
LimitProcessor with multiple input ports is a great idea!

File: dbms/src/Processors/LimitTransform.h [hunk 1]
```diff
@@ -6,48 +6,66 @@
 namespace DB
 {
 
+/// Implementation for LIMIT N OFFSET M
+/// This processor support multiple inputs and outputs (the same number).
+/// Each pair of input and output port works independently.
+/// The reason to have multiple ports is to be able to stop all sources when limit is reached, in a query like:
+///     ELECT * FROM system.numbers_mt WHERE number = 1000000 LIMIT 1
```
Comment (alexey-milovidov) [inline dbms/src/Processors/LimitTransform.h]: Thank you for the comment!

But there is one typo :)

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
> Performance — 1 faster

:+1:

**Pull request merged**